$NetBSD: patch-migemo.rb.in,v 1.2 2025/01/19 11:24:44 taca Exp $

* Add specifying script encoding to EUC-JP.
* Use modern regexp.

--- migemo.rb.in.orig	2003-05-28 12:00:52.000000000 +0000
+++ migemo.rb.in
@@ -1,3 +1,4 @@
+# coding: euc-jp
 #
 # Ruby/Migemo - a library for Japanese incremental search.
 #
@@ -10,11 +11,11 @@
 #
 # NOTE: Ruby/Migemo can work only with EUC_JP encoding. ($KCODE="e")
 #
+Encoding.default_external = Encoding::EUC_JP
 
 require 'migemo-dict'
 require 'migemo-regex'
 require 'romkan'
-require 'jcode'
 include MigemoRegex
 
 class String
@@ -24,7 +25,7 @@ class String
   end
 
   def quotemeta
-    self.gsub(/([^ \w])/, '\\\\\\1')
+    self.gsub(/([[:punct:]])/, '\\\\\\1')
   end
 
   def first
